proposals/0278-screenmanager-layout-management.md

```diff
@@ -11,33 +11,78 @@ Adds a screen manager method for changing the current layout to a new layout.
 ## Motivation
 The manager layer should be able to handle most tasks that used to be handled by RPCs, and the Screen Manager in particular should be able to handle manipulating the screen and templates without requiring the developer to fall back to RPCs in any cases.
 
-One big area that it is currently missing is the ability to manage changing templates, currently done with the `SetDisplayLayout` RPC (on RPC spec <6.0) and `Show` RPC (on RPC spec >6.0). This proposal adds a way to change the layout while taking care of backward compatibility concerns.
+One big area that it is currently missing is the ability to manage changing templates, currently done with the `SetDisplayLayout` RPC (on RPC spec <6.0) and `Show` RPC (on RPC spec >=6.0). This proposal adds a way to change the layout while taking care of backward compatibility concerns.
 
 ## Proposed solution
-We will add a new screen manager method for changing the current layout to a new layout.
+We will add a new screen manager method for changing the current layout to a new layout. This also includes updating the template's night and day color schemes when the feature is available on systems supporting RPC spec 5.0.0 or newer. 
 
 ### iOS APIs
 ```objc
 @interface SDLScreenManager : NSObject
 
-/// Change the current layout to a new layout. You can get a list of SDL-defined layout strings in SDLPredefinedLayout.
-- (void)changeLayout:(NSString *)newLayout withCompletionHandler:(SDLScreenManagerUpdateCompletionHandler)handler;
+/// Change the current layout to a new layout, update the layout's night and day color schemes and update the text, graphics, buttons and template title.
+- (void)changeLayout:(SDLPredefinedLayout)newLayout templateUpdates:(nullable SDLTemplateUpdates *)templateUpdates withUpdateCompletionHandler:(SDLScreenManagerUpdateCompletionHandler)handler;
```

Comment:
This can't just use predefined layout or else custom templates won't work. See below requested updates.